src/components/SchedulePanel/index.js

```diff
@@ -0,0 +1,361 @@
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import { gordonColors } from '../../theme';
+import HoursDialog from './components/OfficeHoursDialog';
+import RemoveHoursDialog from './components/RemoveHoursDialog';
+import EditDescriptionDialog from './components/EditDescriptionDialog';
+import TimeAgo from 'react-timeago';
+
+import schedulecontrol from './../../services/schedulecontrol';
+
+import './schedulepanel.css';
+import myschedule from '../../services/myschedule';
+
+import GordonLoader from '../../components/Loader';
+
+const styles = {
+  colorSwitchBase: {
+    color: gordonColors.neutral.lightGray,
+    '&$colorChecked': {
+      color: gordonColors.primary.cyan,
+      '& + $colorBar': {
+        backgroundColor: gordonColors.primary.cyan,
+      },
+    },
+  },
+  colorBar: {},
+  colorChecked: {},
+};
+
+class GordonSchedulePanel extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      myProf: false, //if my profile page
+      isSchedulePrivate: 0,
+      isExpanded: Boolean,
+      officeHoursOpen: false,
+      disabled: true,
+      selectedEvent: null,
+      isDoubleClick: false,
+      description: '',
+      modifiedTimeStamp: null,
+      loading: true,
+      start:'08:00',
+      end:'17:00',
+      resourceId: 0,
+    };
+    this.scheduleControlInfo = null;
+
+    this.handleIsExpanded = this.handleIsExpanded.bind(this);
+    this.handleOfficeHoursOpen = this.handleOfficeHoursOpen.bind(this);
+    this.handleOfficeHoursClose = this.handleOfficeHoursClose.bind(this);
+    this.handleRemoveOfficeHoursOpen = this.handleRemoveOfficeHoursOpen.bind(this);
+    this.handleRemoveOfficeHoursClose = this.handleRemoveOfficeHoursClose.bind(this);
+    this.handleRemoveButton = this.handleRemoveButton.bind(this);
+    this.handleEditDescriptionOpen = this.handleEditDescriptionOpen.bind(this);
+    this.handleEditDescriptionClose = this.handleEditDescriptionClose.bind(this);
+    this.handleEditDescriptionButton = this.handleEditDescriptionButton.bind(this);
+    this.handleDoubleClick = this.handleDoubleClick.bind(this);
+    this.handleRemoveSubmit = this.handleRemoveSubmit.bind(this);
+  }
+
+  componentWillMount() {
+    this.loadData(this.props.profile);
+    this.setState({ isExpanded: true });
+  }
+
+  loadData = async searchedUser => {
+    try {
+      const scheduleControlPromise = schedulecontrol.getScheduleControl(searchedUser.AD_Username);
+      const scheduleControlInfo = await scheduleControlPromise;
+      this.scheduleControlInfo = scheduleControlInfo;
+    } catch (e) {
+      this.setState({ loading: false });
+    }
+    if (this.scheduleControlInfo) {
+      this.setState({ isSchedulePrivate: this.scheduleControlInfo.IsSchedulePrivate });
```

Comment:
Instead of doing three separate setState calls in this If block, why not save three variables and pass them all in a single setState?

Reply:
Change made